bindings/electron/src/lib.rs

```diff
@@ -10,6 +10,37 @@ lazy_static::lazy_static! {
         Mutex::new(tokio::runtime::Runtime::new().expect("Cannot start tokio runtime for libparsec"));
 }
 
+static BACKTRACE: Mutex<Option<Backtrace>> = Mutex::new(None);
+
+macro_rules! capture_backtrace {
+    ($tx: ident, $($tt: tt)*) => {
+        {
+            std::panic::set_hook(Box::new(|_| {
+                *crate::BACKTRACE.lock().expect("Mutex is poisoned") = Some(std::backtrace::Backtrace::capture());
+            }));
+
+            let ret = $($tt)*;
+
+            let _ = $tx.send(());
```

Comment:
Instead of using the channel for synchronization why not sending the backtrace in ?

Reply:
And in the case there is no error ?

Reply:
If send is success, there is no error

Reply:
I'm not sure to understand, did you mean that we send a message in the channel only when there is a backtrace to be send ?

Reply:
I mean why do you want to send the backtrace in ?

Reply:
If the code reachs the `send` statement, the function hasn't panicked, so why do you want to send the backtrace ?

Reply:
How ?

Reply:
I mean `tx` can't be in the hook

Reply:
`tx` can't be move into the closure ?

Reply:
The hook has `static` lifetime, `tx` don't live enough

Reply:
[`Sender`](https://doc.rust-lang.org/std/sync/mpsc/struct.Sender.html) don't have a lifetime constrain, I'm missing something ?

Reply:
`tx` lives in his current scope (when it has been created) since we don't return it.

Reply:
So I've tried on my side and rust complain about `Sender<Option<Backtrace>>` not implementing `Send` even tho we have:

- `impl<T: Send> Send for Sender<T>`
- `impl<T: Send> Send for Option<T>`
- `impl Send for Bracktrace`